core/src/main/java/com/linecorp/armeria/client/retry/FibonacciBackoff.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.retry;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.common.base.MoreObjects;
+
+final class FibonacciBackoff extends AbstractBackoff {
+    private final long initialDelayMillis;
+    private final long maxDelayMillis;
+
+    FibonacciBackoff(long initialDelayMillis, long maxDelayMillis) {
+        checkArgument(initialDelayMillis >= 0,
+                      "initialDelayMillis: %s (expected: >= 0)", initialDelayMillis);
+        checkArgument(initialDelayMillis <= maxDelayMillis, "maxDelayMillis: %s (expected: >= %s)",
+                      maxDelayMillis, initialDelayMillis);
+        this.initialDelayMillis = initialDelayMillis;
+        this.maxDelayMillis = maxDelayMillis;
+    }
+
+    @Override
+    protected long doNextDelayMillis(int numAttemptsSoFar) {
+        final long nextDelay = saturatedMultiply(initialDelayMillis, fib(numAttemptsSoFar));
+        return Math.min(nextDelay, maxDelayMillis);
+    }
+
+    private static int fib(int n) {
+        int a = 1;
```

Comment:
Instead of calculating a multiplier, can we do addition using nextDelay directly? It would make it easy to cap the calculation at maxDelay.

If I not mistaken, as the number of attempts goes up this calculation will take longer and longer, unbounded.

Reply:
Thanks for the review! Other solutions I thought of is by keeping the last two multipliers in fields. As this is in the client package, I assume that the backoff instance is not shared, but I would have to read more of the code to make sure.

Or another solution would be creating a look-table upon initialization, it only has to have a size of 90-something for long.

Reply:
A lookup table sounds nice, since in practice, it'll probably be far fewer, like 5 entries. We may as well do that for ExponentialBackoff too :D